### PR TITLE
Changed 'buildcache list' to more closely resemble 'spack find'

### LINKED ISSUE #6331: Spack buildcache is too verbose
For a single package: 
``` bash
# spack buildcache list
==> Finding buildcaches in <MIRROR_ROOT>/build_cache
==> Fetching file://<MIRROR_ROOT>/build_cache/linux-centos7-x86_64-gcc-4.8.5-xextproto-7.3.0-v5vttf7jlxd4lkqhrtc4x5zflbudgpov.spec.yaml
######################################################################## 100.0%
==> Enter
spack buildcache install /v5vttf7
 to install "xextproto@7.3.0%gcc@4.8.5 arch=linux-centos7-x86_64 "
```

```buildcache list``` will print two lines for each new spec discovered, then three more lines for each package. I think a better output would look similar to ```spack find``` output. The download output isn't useful for someone that wants a list of available packages.

``` bash
# spack find -lv --show-full-compiler zlib
==> 1 installed packages.
-- linux-centos7-x86_64 / gcc@4.8.5 -----------------------------
64vg6e4 zlib@1.2.11%gcc@4.8.5+optimize+pic+shared
```

### PR DESCRIPTION
**Pull request by sknigh**
Fixes #6331 

```spack buildcache list``` has the following new flags:
``` bash
--show-flags          show spec compiler flags
--show-full-compiler  show full compiler specs
-v, --variants        show variants in output (can be long)
```

Output
``` bash
# spack buildcache list apr-util
==> Finding buildcaches in <MIRROR>/build_cache
==> buildcache spec(s) matching 'apr-util'
-- linux-centos7-x86_64 / gcc@4.8.5 -----------------------------
ev3wtha apr-util@1.6.0

# spack buildcache list -v apr-util
==> Finding buildcaches in <MIRROR>/build_cache
==> buildcache spec(s) matching 'apr-util'
-- linux-centos7-x86_64 / gcc@4.8.5 -----------------------------
ev3wtha apr-util@1.6.0+crypto~gdbm~odbc~pgsql~sqlite
```

### CODE REVIEW

**Comment by sknigh** [comment]
The build tests are timing out on Python 2.7.13, yet they finish on my system.

File: lib/spack/spack/cmd/buildcache.py [hunk 1]
```diff
@@ -80,6 +80,24 @@ def setup_parser(subparser):
     listcache.add_argument(
         'packages', nargs=argparse.REMAINDER,
         help="specs of packages to search for")
+
+    listcache.add_argument('--show-flags',
```
Comment (tgamblin) [inline lib/spack/spack/cmd/buildcache.py]: use common arguments like #6343.

**Comment by tgamblin** [comment]
LGTM but now it's conflicted -- @sknigh can you resolve?

@gartung: do you want to comment on this?

**Comment by gartung** [comment]
Both #6343 and #10052 produce the same output. I would like to see at least something that mentions you can download by hash listed by spack buildcache list -l.

**Comment by sknigh** [comment]
Closing in favor of #10052

**Pull request closed without merging**